src/sph-layout.cc

```diff
@@ -0,0 +1,225 @@
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+/// \file
+/// \brief Print the layout of partitions for the specified
+///        configuration.
+
+#include <iostream>
+#include <fstream>
+#include <sstream>
+#include <stdexcept>
+#include <string>
+#include <utility>
+#include <vector>
+#include <map>
+#include <cmath>
+
+#include "boost/program_options.hpp"
+#include "boost/shared_ptr.hpp"
+
+#include "lsst/partition/Chunker.h"
+
+#include "lsst/partition/Geometry.h"
+
+namespace po    = boost::program_options;
+namespace part  = lsst::partition;
+
+namespace {
+
+    // An association between chunk identifiers and the corresponidng worker nodes
+    // where chunks are deployed.
+    typedef std::map<int32_t, std::string> Chunk2WorkerMap;
+
+    // A parser for the chunk-to-worker map file.
+    Chunk2WorkerMap parseChunk2WorkerMap(std::string const & filename) {
+
+        Chunk2WorkerMap result;
+
+        std::ifstream infile {filename,  std::ifstream::in};
+        for (std::string line; std::getline(infile, line);) {
+
+            std::stringstream is(line);
+            int32_t chunk;
+            std::string node;
+            is >> chunk;
+            is >> node;
+            
+            result[chunk] = node;
+        }
+        return result;
+    }
+
+    // Dump the header of the output table
+    void dumpHeader() {
+
+        std::cout
+            << "\n"
+            << "        |        RA [degree]          |      DECL [degree]          |              |            \n"
+            << "     id |--------------+--------------+--------------+--------------|    Area [sr] |     Worker \n"
+            << "        |          Min |          Max |          Min |          Max |              |            \n"
+            << " -------+--------------+--------------+--------------+--------------+--------------+------------\n";
+    }
+
+    // Dump the footer of the output table
+    void dumpFooter() {
+
+        std::cout
+            << "\n";
+    }
+
+    // Dump a row of the output table
+    void dumpRow(
+        int32_t            const   chunkId,
+        part::SphericalBox const & box,
+        Chunk2WorkerMap    const & chunk2worker) {
+
+        const double area = box.area();
```

Comment:
Of course standard does not care. But we do :smile:  I'm just saying that you should pick one style and use it consistently for the whole file, mixing styles makes it harder to read. Our style guide suggest that `type const` is preferred over `const type`.